exchanges/poloniex/poloniex_websocket.go

```diff
@@ -87,196 +94,318 @@ func (p *Poloniex) WsHandleData() {
 		select {
 		case <-p.Websocket.ShutdownC:
 			return
-
 		default:
 			resp, err := p.WebsocketConn.ReadMessage()
 			if err != nil {
 				p.Websocket.ReadMessageErrors <- err
 				return
 			}
 			p.Websocket.TrafficAlert <- struct{}{}
-			var result interface{}
-			err = json.Unmarshal(resp.Raw, &result)
+			err = p.wsHandleData(resp.Raw)
 			if err != nil {
 				p.Websocket.DataHandler <- err
-				continue
 			}
-			switch data := result.(type) {
-			case map[string]interface{}:
-				// subscription error
-				p.Websocket.DataHandler <- errors.New(data["error"].(string))
-			case []interface{}:
-				chanID := int(data[0].(float64))
-				if len(data) == 2 && chanID != wsHeartbeat {
-					if checkSubscriptionSuccess(data) {
-						if p.Verbose {
-							log.Debugf(log.ExchangeSys,
-								"%s websocket subscribed to channel successfully. %d",
-								p.Name,
-								chanID)
+		}
+	}
+}
+
+func (p *Poloniex) wsHandleData(respRaw []byte) error {
+	var result interface{}
+	err := json.Unmarshal(respRaw, &result)
+	if err != nil {
+		return err
+	}
+	if data, ok := result.([]interface{}); ok {
+		if len(data) == 0 {
+			return errors.New("c'mon man, no data")
```

Comment:
😆 in this case, we can just return nil as the exchange didn't send us any meaningful data